.github/workflows/ci.yml

```diff
@@ -39,20 +39,31 @@ jobs:
       - name: Run ShellCheck
         uses: ludeeus/action-shellcheck@master
         with:
-            ignore_paths: ./.devcontainer
+          ignore_paths: ./.devcontainer
 
   test-install:
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        shell: [sh, bash, zsh]
+        shell: [sh, bash, zsh, fish]
     runs-on: ${{ matrix.os }}
     steps:
       - name: Print build information
         run: "echo head_ref: ${{ github.head_ref }}, ref: ${{ github.ref }}"
 
       - uses: actions/checkout@v3
 
-      - name: "Run with ${{ matrix.shell }}"
-        run: sh -c '${{ matrix.shell }} -c "./install.sh"'
+      - name: Install ${{matrix.shell}} using apt-get
+        if: matrix.os == 'ubuntu-latest' && matrix.shell != 'sh'
+        run: "sudo apt-get install -y ${{ matrix.shell }}"
+
+      - name: Install ${{matrix.shell}} using Homebrew
+        if: matrix.os == 'macos-latest'  && matrix.shell != 'sh'
+        run: "brew update && brew install ${{ matrix.shell }}"
+
+      - name: Update SHELL env variable
+        run: "env SHELL=$(which ${{ matrix.shell }})"
```

Comment:
Should we be using https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell instead?

Reply:
i've tried this though have been running into syntax issues with `shell: ${{ .. }}` approach. Hardcoding the shells worked